### PR TITLE
Work on deep_conmech and code cleanup

### PR DESCRIPTION
**Pull request by michalJur**

### CODE REVIEW

File: conmech/dataclass/body_coeff.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 
 @dataclass
-class Coefficients:
+class BodyCoeff:
```
Comment (piotrbartman) [inline conmech/dataclass/body_coeff.py:6]: I porpose `BodyProperties`

File: conmech/dataclass/obstacle_coeff.py [hunk 2]
```diff
@@ -0,0 +1,7 @@
+from dataclasses import dataclass
+
+
+@dataclass
+class ObstacleCoeff:
```
Comment (piotrbartman) [inline conmech/dataclass/obstacle_coeff.py:5]: `ObstacleProperties`

File: conmech/dataclass/time_data.py [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+
+@dataclass
+class TimeData:
+    final_time: float
+    time_step: float = 0.01
+
+    @property
+    def episode_steps(self):
+        return int(self.final_time / self.time_step)
```
Comment (piotrbartman) [inline conmech/dataclass/time_data.py:12]: lack of new line at the end of file

File: conmech/dataclass/time_data.py [hunk 4]
```diff
@@ -0,0 +1,12 @@
+from dataclasses import dataclass
+from typing import Optional
+
+
+@dataclass
+class TimeData:
```
Comment (piotrbartman) [inline conmech/dataclass/time_data.py:6]: I propose `Schedule`

File: conmech/helpers/cmh.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    return os.listdir(directory)
+
+def clear_folder(directory):
+    if not os.path.exists(directory):
+        return
+    shutil.rmtree(directory)
+
+def recreate_folder(directory):
+    clear_folder(directory)
+    create_folder(directory)
```
Comment (piotrbartman) [inline conmech/helpers/cmh.py:44]: new line

File: deep_conmech/simulator/setting/setting_obstacles.py [hunk 6]
```diff
@@ -211,40 +274,66 @@ def normalized_obstacle_normal(self):
...
+            self.boundary_v_tangential,
+            self.obstacle_coeff.friction,
+            self.time_step,
+        )
 
     @property
-    def normalized_boundary_obstacle_penetration_vectors(self):
-        return self.normalize_rotate(self.boundary_obstacle_penetration_vectors)
+    def is_coliding(self):
+        return np.any(self.boundary_penetration_norm > 0)
```
Comment (piotrbartman) [inline deep_conmech/simulator/setting/setting_obstacles.py:339]: new line

File: deep_conmech/simulator/setting/setting_obstacles.py [hunk 7]
```diff
@@ -211,40 +274,66 @@ def normalized_obstacle_normal(self):
...
-        return self.boundary_obstacle_penetration * self.boundary_obstacle_normals
+    def boundary_penetration(self):
+        boundary_penetration_norm = self.boundary_penetration_norm
+        return boundary_penetration_norm * self.boundary_obstacle_normals
+
+    @property
+    def normalized_boundary_penetration(self):
+        return self.normalize_rotate(self.boundary_penetration)
+
+    #######
```
Comment (piotrbartman) [inline deep_conmech/simulator/setting/setting_obstacles.py:310]: unnecessary comment

File: conmech/dataclass/obstacle_properties.py [hunk 8]
```diff
@@ -2,6 +2,6 @@
```
Comment (piotrbartman) [inline conmech/dataclass/obstacle_properties.py:2]: please rename file to `obstacle_properties.py`

Comment (michalJur) [reply conmech/dataclass/obstacle_properties.py:2]: both files are renamed, is there a typo?

File: conmech/dataclass/time_data.py (hunk 4 above)
Comment (michalJur) [reply conmech/dataclass/time_data.py]: Agreed and changed

**Pull request merged**